test/test_performances.jl

```diff
@@ -0,0 +1,131 @@
+function DoTest()
+    sleep(0.1)
+    println("Is (...) a perf-killer if we used with collect?")
+    function foo1(args...)
+        for x in args
+            i = x
+        end
+    end
+    function foo2(arg_coll)
+        for x in arg_coll
+            i = x
+        end
+    end
+    arg_coll = 1:50
+    @time foo1(arg_coll...)
+    @time foo2(arg_coll) # much better
+    println()
+
+    println("Is (...) a perf-killer if used often with 1 arg?")
+    function bar1(x...)
+        x[1] + 1
+    end
+
+    function bar2(x)
+        x + 1
+    end
+    x = 5
+    @time bar1(x)
+    @time bar2(x) # Better
+    println()
+
+    println("map vs generators with eachindex")
+    n = 10
+    x = Tuple(rand(n))
+    y = Tuple(rand(n))
+    v = Tuple(rand(n))
+    @time Tuple(round.(Int, x .+ v.*y)) # better
+    @time Tuple(round(Int, x[i] + v[i]*y[i]) for i in eachindex(x))
+    println()
+
+    println("Comparison UnitRange, expand..., and a:b")
+    a = 20
+    b = 100
+    lb = collect(a:b)
+    ub = collect(a+20:b+20)
+    @time Tuple(UnitRange(x...) for x in zip(lb, ub))
+    @time Tuple(lb[i]:ub[i] for i in eachindex(lb))
+    @time Tuple(UnitRange(lb[i],ub[i]) for i in eachindex(lb)) # slightly better
```

Comment:
I think https://github.com/blegat/Dionysos.jl/pull/9/files#diff-44d1eb315bf0cecb0c500e4285950101R35-R37
is even better

Reply:
Do you mean this:?
```
function _make_iterator_from_lims(rect)
    return (collect(pos) for pos in Iterators.product((UnitRange(x...) for x in zip(rect.lb, rect.ub))...))
    return (SVector(pos...) for pos in Iterators.product(_ranges(rect)...))
end
```
If I am not wrong the output is not type-stable (I double check and come back)

Reply:
Aaaaah si tu fixes la longueur de _range (comme tu le fais) ce n'est pas type-unstable...

Reply:
Yes, exactly, in fact we should do `ntuple(..., Val(N))` instead of `ntple(..., N)`, see https://github.com/JuliaLang/julia/blob/master/base/ntuple.jl#L50-L55